ios/RCTWebRTC/WebRTCModule+RTCMediaStream.m

```diff
@@ -268,12 +265,39 @@ - (void)getUserVideo:(NSDictionary *)constraints
   }
 
   if (videoDevice) {
-    // TODO: Actually use constraints...
-    RTCAVFoundationVideoSource *videoSource = [self.peerConnectionFactory avFoundationVideoSourceWithConstraints:[self defaultMediaStreamConstraints]];
-    // FIXME The effort above to find a videoDevice value which satisfies the
-    // specified constraints was pretty much wasted. Salvage facingMode for
-    // starters because it is kind of a common and hence important feature on
-    // a mobile device.
+    NSMutableDictionary *mandatory = [[NSMutableDictionary alloc]init];
+    if (videoConstraints != nil) {
+      NSDictionary *mandatoryParameters = videoConstraints[@"mandatory"];
+      if (mandatoryParameters != nil) {
+        if ([mandatoryParameters valueForKey:@"maxFrameRate"] != nil) {
+          NSMutableString *maxFrameRate = mandatoryParameters[@"maxFrameRate"];
+          [mandatory setObject:maxFrameRate forKey:@"maxFrameRate"];
+        }
+        if ([mandatoryParameters valueForKey:@"maxHeight"] != nil) {
+          NSMutableString *maxHeight = mandatoryParameters[@"maxHeight"];
+          [mandatory setObject:maxHeight forKey:@"maxHeight"];
+        }
+        if ([mandatoryParameters valueForKey:@"maxWidth"] != nil) {
+          NSMutableString *maxWidth = mandatoryParameters[@"maxWidth"];
+          [mandatory setObject:maxWidth forKey:@"maxWidth"];
+        }
+        if ([mandatoryParameters valueForKey:@"minFrameRate"] != nil) {
+          NSMutableString *minFrameRate = mandatoryParameters[@"minFrameRate"];
+          [mandatory setObject:minFrameRate forKey:@"minFrameRate"];
+        }
+        if ([mandatoryParameters valueForKey:@"minHeight"] != nil) {
+          NSMutableString *minHeight = mandatoryParameters[@"minHeight"];
+          [mandatory setObject:minHeight forKey:@"minHeight"];
+        }
+        if ([mandatoryParameters valueForKey:@"minWidth"] != nil) {
+          NSMutableString *minWidth = mandatoryParameters[@"minWidth"];
+          [mandatory setObject:minWidth forKey:@"minWidth"];
+        }
+      }
+    }
```

Comment:
you did not use `defaultMediaStreamConstraints` at all.

if `mandatory` is not touched at the end, `mandatory` should fallback to `defaultMediaStreamConstraints`  for compatibility.